snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -347,7 +347,7 @@ public FileBasedReceivedSnapshot newReceivedSnapshot(final String snapshotId) {
   }
 
   private void addPendingSnapshot(final PersistableSnapshot pendingSnapshot) {
-    actor.call(() -> pendingSnapshots.add(pendingSnapshot));
+    actor.run(() -> pendingSnapshots.add(pendingSnapshot));
```

Comment:
:x: You can use submit now even from outside the actor, and it guarantees you get append semantics (and not prepend when you call it from an actor thread, as with run). The one caveat is that submitted tasks via submit will not be ran during the closing phase, whereas those submitted with `run` will. But that was already the case with `call`, so replacing `call` with `submit` is safe.

Reply:
Thanks. Didn't realize "submit" has a new semantics.